### ISSUE TITLE
Sitemap generation fails because of missing image info

### ISSUE THREAD

**Issue by OvalMedia** [mageplaza/magento-2-blog#295]
### Preconditions (*)

1. Magento 2.4.1
2. PHP 7.3

### Steps to reproduce (*)

1. Create blog posts with image(s)
2. Generate sitemap

### Expected result (*)

1. Sitemap gets generated with all products, categories and blog posts
2.

### Actual result (*)

1. Generation fails with error "TypeError: Argument 1 passed to Magento\Sitemap\Model\Sitemap::escapeXmlText() must be of the type string, null given, called in /vendor/magento/module-sitemap/Model/Sitemap.php on line 576 and defined in /vendor/magento/module-sitemap/Model/Sitemap.php:598"

In magento's function _getSitemapRow (\vendor\magento\module-sitemap\Model\Sitemap.php) the $images object is supposed to contain a title which is passed to escapeXmlText(). 

I created a dump on each item to see what's wrong. This is an image (data object) of a product:

```
Array
(
    [collection] => Array
        (
            [0] => Magento\Framework\DataObject Object
                (
                    [_data:protected] => Array
                        (
                            [url] => https://www.example.com/media/catalog/product/cache/4014ca3c284ee8bfa0947c8aaa649e60/t/e/test.jpg
                            [caption] => 
                        )

                )

        )

    [title] => My awesome product
    [thumbnail] => https://www.example.com/media/catalog/product/cache/4014ca3c284ee8bfa0947c8aaa649e60/t/e/test.jpg
)
```
This is an image of a blog post:

```
Array
(
    [collection] => Array
        (
            [0] => Magento\Framework\DataObject Object
                (
                    [_data:protected] => Array
                        (
                            [url] => https://www.example.com/media/mageplaza/blog/post/t/e/test.jpg
                            [caption] => 
                        )

                )

        )

)
```

It is missing the title property which results in $images->getTitle() to return NULL instead of a string. escapeXmlText expects a string, which is why it fails.

**Comment by phutx** [comment]
Hi @OvalMedia ,
Please download this file and replace it with the path:
app/code/Mageplaza/Blog/Model/Sitemap.php
[Sitemap.zip](https://github.com/mageplaza/magento-2-blog/files/5464145/Sitemap.zip)

**Comment by OvalMedia** [comment]
I'd rather have this fixed in the repo since it is managed via composer inside the vendor folder.

**Comment by phutx** [comment]
Hi,
So you just need replace it with the path:
vendor/mageplaza/Blog/Model/Sitemap.php

**Comment by OvalMedia** [comment]
I understand. But I manage the code with composer. Another "composer update" and the replacement is gone.

**Comment by vinylcutgraphics2** [comment]
Does not work for me - any update?

**Comment by ithubdeveloper** [comment]
Hi, 
You can update the file using Magento 2 patches technique

Please follow these steps.

https://www.classyllama.com/blog/create-apply-patches-magento-2

**Comment by OvalMedia** [comment]
This is months old. Why can't you just fix it and push it into the repo?

**Comment by ithubdeveloper** [comment]
@OvalMedia ,

I'm also facing the same issue but that file did not work for me, I just suggest a technique to modify the vendor file using patches.

If you have found out a solution can you please share here?

**Comment by vinylcutgraphics2** [comment]
I gave up with th Mageplaza on this isuue, they had fixed it for me before, but lose the fix every time i upgrade

I fixed using their own sitemap module
https://www.mageplaza.com/magento-2-google-sitemap/

Seems to generate OK now

**Comment by anvanza** [comment]
Why is there even a DI preference in the first place? This is an overwrite of the sitemap. Instead there should be a decent solution by adding an ItemProvider.

**Comment by anvanza** [comment]
Anyway, I created a PR that works for me.

**Comment by Henry17888** [comment]
when I generate a site map via marketing/seo&search/site map/generate, I also get this error. 
Magento version: V2.4.2-p1
blog version: 4.1.0
so, please kindly advise how to fix this?

**Comment by roman204** [comment]
This error is resolved with version 4.1.3
